pkg/stack/tile_stack.go

```diff
@@ -0,0 +1,66 @@
+package stack
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+)
+
+type Tile struct { // To be replaced by #9
+	id int
+}
+
+type Stack struct {
+	seed    int64
+	turn_no int32
+	tiles   []Tile
+	order   []int32
+}
+
+func New(tiles []Tile) Stack {
+	return NewSeeded(tiles, time.Now().UnixNano())
+}
+
+func NewSeeded(tiles []Tile, seed int64) Stack {
+	stack := Stack{
+		seed:    seed,
+		turn_no: 0,
+		tiles:   tiles,
+		order:   make([]int32, len(tiles)),
+	}
+	for i := range len(tiles) {
+		stack.order[i] = int32(i)
+	}
+	return stack
+}
+
+func (s *Stack) Shuffle() {
```

Comment:
Why do we need another array of indexes defining order of tiles? Isn't slice alraedy ordered?